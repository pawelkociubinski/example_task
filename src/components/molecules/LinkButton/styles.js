import styled from 'styled-components';

export const A = styled.a`
  display: inline-block;
  text-decoration: none;
  color: inherit;
`;
