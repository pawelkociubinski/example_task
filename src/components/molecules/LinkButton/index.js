/* @flow */
// Dependencies
import type { Node } from 'react';
import React from 'react';
// Custom
import Button from '../../atoms/Button';
import { A } from './styles';

type Props = {
  href: string,
  children: Node,
}

const LinkButton = (props: Props) => {
  const { children, href } = props;
  return (
    <A href={href}>
      <Button>
        {children}
      </Button>
    </A>
  );
};

export default LinkButton;
