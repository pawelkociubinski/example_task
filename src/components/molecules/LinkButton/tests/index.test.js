import React from 'react';
import renderer from 'react-test-renderer';
import LinkButton from '../index';

describe('<LinkButton />', () => {
  it('renders correctly', () => {
    const component = renderer.create(<LinkButton href="www.example.com">example</LinkButton>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
