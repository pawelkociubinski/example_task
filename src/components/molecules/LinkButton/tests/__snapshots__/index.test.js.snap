// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> renders correctly 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #DDDDDD;
  color: #707070;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  overflow: hidden;
  position: relative;
  border: none;
  padding: 0 40px;
}

.c1:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c0 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

<a
  className="c0"
  href="www.example.com"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    example
  </button>
</a>
`;
